src/components/DataInput/inputHandlerFactories.ts

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { SelectChangeEvent } from "@mui/material";
+
+type valueSetter = {
+    (value: any): void;
```

Comment:
is this 'any' not just a `string`? afaik all the inputs you have are string valued